pkg/upgrade/migrate_deprecated_test.go

```diff
@@ -0,0 +1,38 @@
+package upgrade
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+func TestMigrateAllDeprecated(t *testing.T) {
+	// prepare
+	nsn := types.NamespacedName{Name: "my-instance"}
+	existing := *v1.NewJaeger(nsn)
+	existing.Spec.Collector.Options = v1.NewOptions(map[string]interface{}{
+		"migrate-from":  "value-not-migrated",
```

Comment:
nit: "value-not-migrated" - might be better to have "value-to-migrate" - as implies it shouldn't be migrated.

Reply:
Interesting, in my mind, the value isn't being migrated, but I think your perspective is more accurate: the old value is being migrated to a new key.